### DISCUSSION TITLE
Code for Adaptive/Practical Quad Mesh Simplification

### DISCUSSION THREAD

**Discussion by SeyedAlirezaFatemi** [cnr-isti-vclab/meshlab#1056] [Q&A]
Hello.
I've searched the code, but I can't find where the simplification methods described in the paper "Adaptive Quad Mesh Simplification" or "Practical Quad Mesh Simplification" are located. I wonder if the implementation of these papers exists in meshlab or anywhere else over the internet. If you know where I can find this code, please let me know.
I'm asking this here because this was mentioned in the paper.
![image](https://user-images.githubusercontent.com/39184114/128591447-77e44318-c8c3-4eb6-b844-c3adeab8e3c2.png)
Thanks!

**Comment by alemuntoni** [comment]
Hi, 
I actually don't know if this has been added as meshlab plugin in the end.
@nicopietroni do you know something about this?

**Reply by Seangongx** [reply]
Hello,
I tried to find the two relevant functions and use them but failed too.
Some people in Youtube said:  
"it is under Filters > Remeshing, reconstruction, simplification > Simplification: MC Edge Collapse"
Could you please help us for checking them exist or not? And the Q&A could be solved.
Thanks a lot.

**Reply by alemuntoni** [reply]
I guess that "Simplification: MC Edge Collapse" at some point become "Simplification: Edge Collapse for Marching Cubes meshes".

**Reply by cignoni** [reply]
The _Simplification: Edge Collapse for Marching Cubes meshes_ filter is not related to quad meshing

**Reply by Seangongx** [reply]
> I guess that "Simplification: MC Edge Collapse" at some point become "Simplification: Edge Collapse for Marching Cubes meshes".

Thank you both :-)

**Comment by Seangongx** [comment]
Just put some answers may useful here.

> https://www.meshlab.net/ 

at the bottom have a discription about these functions. I try to search the key word in the repos but found nothing.

> https://stackoverflow.com/questions/62720269/fail-to-save-quad-mesh-in-meshlab

Seems someone found how to use them but I have no idea either.

**Reply by cignoni** [reply]
The quad simplification code in the end was removed for not being very robustly written. You can find it in the vcg library at your own risk :) 
https://github.com/cnr-isti-vclab/vcglib/tree/master/apps/sample/polygonmesh_quadsimpl

**Reply by Seangongx** [reply]
> The quad simplification code in the end was removed for not being very robustly written. You can find it in the vcg library at your own risk :) https://github.com/cnr-isti-vclab/vcglib/tree/master/apps/sample/polygonmesh_quadsimpl

Thanks a million for your attention and quick response!

**Reply by JulianKnodt** [reply]
Sorry to revive an older discussion, but I found that there are numerous compilation errors when building `polygonmesh_quadsimpl`, and I had to modify a number of cmake files to compile it as well. Is there a canonical, buildable version somewhere?
